phpBB/phpbb/storage/adapter/adapter_interface.php

```diff
@@ -95,4 +95,13 @@ public function copy($path_orig, $path_dest);
 	 *
 	 */
 	public function get_link($path);
+
+	/*
+	 * Get space available in bytes.
+	 *
+	 * @throws \phpbb\storage\exception\exception		When unable to retrieve available storage space
+	 *
+	 * @return int	Returns available space
+	 */
+	public function free_space();
```

Comment:
I think there is some cases where you cannot estimate the free space (because you have no limit per example).
How should be handle this case? throw an exception ? return PHP_INT_MAX ?

Reply:
With an exception

Reply:
It can be a problem here https://github.com/phpbb/phpbb/pull/5243
Is there any way to catch an exception with twig?

Reply:
not really :/

Reply:
we could imagine to extend twig for that but I don't think it is a good idea
maybe we could add a parameter to free_space() to not throw but.... don't really like it tbh.
Maybe you will have another idea? or @marc1706 ?

Reply:
Yeah, I'd rather not hack exception treatment into twig. If there is no limit I'd rather return PHP_INT_MAX and maybe handle this accordingly before template output. An exception is IMHO not the correct tool to use here as this is not an anomalous condition but rather an expected side effect of the used storage adapter.

Reply:
I prefer it like this, or return -1 or something, because i have to detect if isn't available from twig, if not it will show a few petabytes available.
Also i am looking and php_int_max is only 2gb

Reply:
PHP_INT_MAX normally is 2147483647 (int 32 bits systems, but much bigger in 64 bits systems)